### ISSUE TITLE
QueueTrigger does not emit errors if the queue doesn't exist

### ISSUE THREAD

**Issue by ogvolkov** [Azure/azure-webjobs-sdk#2498]
If the queue specified in QueueTrigger does not exist, Azure function is not triggered, and no error messages are produced. This makes configuration errors very painful to debug, as there are no exceptions, no log messages - it just doesn't work.

It seems that queue existence is checked in [QueueListener](https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Listeners/QueueListener.cs#L191), but nothing is done if it doesn't exist. It might be better to validate it in [QueueTriggerBinding](https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Triggers/QueueTriggerAttributeBindingProvider.cs#L77) right after getting a reference to the queue, so configuration errors are detected very early in the process. Are there any reasons not to do so?

**Comment by v-anvari** [comment]
Hi @ogvolkov , Apologies for the delayed response, can you provide us the repro steps of what exactly are you trying. Are you creating a queue and then deleting it? or are are you not creating a queue itself. Because for the latter case, usually the function App throws an error saying it doesn't find the queue

**Comment by ogvolkov** [comment]
Hi @v-anvari, here's what I can recollect: it was a Java function, running on 1.3.1 Azure functions java library, and the queue name was incorrect for whatever reason (I don't quite remember why, maybe it wasn't reading it from the environment correctly or there was something with the reserved words, but I can't remember exactly). I'm pretty sure that the function did start normally, but was never ever triggered by the queue and I haven't found any errors in logs.

Somehow I connected it to the source code I've linked (I also tried to debug it locally as far as I remember), and found that it does check for queue existence in some places, but doesn't signal it. This analysis might be wrong, or maybe now there's a check on some higher level, but I cannot verify it as I've moved from using Azure functions since.
